### ISSUE TITLE
Implement the search page

### ISSUE THREAD

**Issue by hellais** [ooni/explorer#12]
Here are some items that still need to be done:
* [ ] Add hints and validation to the ASN input box
* [ ] If another search query is done we should terminate previous queries (see: #20)
* [x] Only list countries where we actually have measurements in 
* [x] Populate the list of test_names from a more canonical source (maybe the API?)

[Link to mockup](https://design.ooni.io/mockups/explorer/screen/e84dd189-cece-4495-9982-637ca77eed2d/Search-1/)

Currently the search page is done in part, however it's not fully aligned to the mockups.

In particular some of the form elements are not the same as the mockups. These components should eventually make their way into the `design-system`.

**Comment by gabelula** [comment]
Related to https://github.com/ooni/explorer/issues/2

**Comment by hellais** [comment]
The search page is for the most part done. 

We leave this open as there are still a couple of final touches that should be done prior to the 2.0.0 beta, namely:
* [ ] Start date & end date filters don't work as expected (the filtering does not work and the address bar is not rewritten)
* [ ] Date filter selectors should validate that the date range is consistent (i.e. start_date < end_date < NOW())
* [ ] Add hints and validation to the ASN input box
* [ ] If another search query is done we should terminate previous queries (see: https://github.com/ooni/explorer/issues/20)
* [ ] Only list countries where we actually have measurements in
* [ ] Populate the list of test_names from a more canonical source (maybe the API?)

I also think the component for the search page could benefit from a bit of refactoring, especially coming up with a better pattern for writing the URL based on the filter options.

It would also be good to get input from @elioqoshi and @holantonela on the overall style and layout of this page (I believe they didn't pay as much attention to this page as to the measurements pages).

**Comment by gabelula** [comment]
@elioqoshi did you have any time to review this one?

**Comment by elioqoshi** [comment]
@gabelula Thanks for bringing this to my attention. I haven't finished my review yet, will wrap it up tomorrow.

**Comment by sarathms** [comment]
Looking into refactoring the search page components to make the URL rewriting more responsive.

**Comment by hellais** [comment]
So to address the items I have assigned to myself, I have now implements bits of this as part of: https://github.com/ooni/api/pull/77

> Only list countries where we actually have measurements in (@hellais will document how to do this)

This can be done by fetching the API endpoint: http://api.test.ooni.io/api/_/countries

> Populate the list of test_names from a more canonical source (maybe the API?) (@hellais will document how to do this)

This can be obtained via: http://api.test.ooni.io/api/_/test_names (this actually already existed)

It may be useful to add also test_group support to this endpoint, but we can do that at a later stage once we actually need to cluster them by groups.

**Comment by agrabeli** [comment]
In the Search page, we display colour-coding (red, green, orange, yellow) based on specific types of results ("blocked", "reachable", "anomaly", "error") which only really apply to the Web Connectivity test (though they also apply, to some extent, to the IM tests as well). 

Since we're also displaying numerous other tests as well (which render completely different types of results), I propose the following changes to the table (with the results):

* [ ] Display "orange" or "green" (on the left-side of the table) for the **IM test (WhatsApp, Facebook Messenger, Telegram)** results depending on whether the apps are "blocked ("orange") or reachable ("green"). => I propose using "orange" (anomaly) because we cannot confirm blocking 100% with a single measurement.

* [ ] Display "orange" or "green" (on the left-side of the table) for **HTTP Invalid Request Line** + **HTTP Header Field Manipulation** test results depending on whether a middlebox is detected ("orange") or not detected ("green").

* [ ] Do not display any colours (on the left-side of the table) for the NDT, DASH, and all other legacy tests (e.g. DNS injection, Psiphon, Captive Portal, etc.) because we haven't applied heuristics for displaying processed results (yet).

What do you think?

**Comment by sarathms** [comment]
- [ ] Search results should be linked with the same schema that is used in the legacy explorer results.

**Comment by sarathms** [comment]
@agrabeli, I think the first and second checklist items you mentioned [here](https://github.com/ooni/explorer/issues/12#issuecomment-475716307) are already done. But I will verify again and check them off.

I am a bit unsure about 
> Delete "Web Connectivity" from the results and only display the tested URL

It seems to make the columns a bit inconsistent. I agree that we should shove the URL in that column. Maybe we should use the last column as a miscellaneous one and put it there. Maybe @hellais  @elioqoshi @holantonela also have opinions on this.

**Comment by elioqoshi** [comment]
> Delete "Web Connectivity" from the results and only display the tested URL

I don't think it's ideal, but it's a quick solution. Wouldn't it look the same as HTTP Requests then?

**Comment by hellais** [comment]
> Wouldn't it look the same as HTTP Requests then

Yeah, but I don't think it matters. A user of search cares about a result pertaining to the accessibility of a website, I don't think they necessarily care too much if it was gathered with a web connectivity or a HTTP Request test (moreover HTTP Request is deprecated since more than 2 years now so it will show up less and less frequently in the results).

**Comment by elioqoshi** [comment]
Then let's do it. 

I do think that the alignment and layout of elements in a search result row needs to be tweaked as well at some point but I leave triaging this to you.

**Comment by sarathms** [comment]
> Add hints and validation to the ASN input box

@hellais This is the only one that needs to be checked before this issue can be closed. I think we should add a `tooltip` prop to all the input components which provides a `PropType.node` shown on hovering a `(?)` icon near the label. We should be able to provide links to relevant documentation about nettests, spec etc.

What kind of validation can be done on the AS input?

**Comment by hellais** [comment]
> What kind of validation can be done on the AS input?

I guess we can check that the format is either just a number or a number with the `AS` prefix. If it's just a number I think we need to prepend `AS` to it before doing the search.

**Comment by sarathms** [comment]
Looks like you get the same results when you send in `1234` or `AS1234`. So we can accept one of the two formats and reject others.
